src/qibo/callbacks.py

```diff
@@ -119,20 +125,23 @@ def nqubits(self, n: int):
             ]
 
     def apply(self, backend, state):
-        rho = backend.partial_trace(state, self.partition, self.nqubits)
-        entropy, spectrum = backend.entanglement_entropy(rho)
+        from qibo.quantum_info.metrics import entanglement_entropy
```

Comment:
This can also be top level.

Reply:
at top level it creates a circular import error